haskell/private/runghc.bzl

```
@@ -0,0 +1,38 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
+
+# Note [Running Setup.hs when cross-compiling]
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# When cross compiling we interpret Setup.hs in the execution
+# platform. In order to achieve this, we provide a runghc of a
+# toolchain targeting the execution platform.
+#
+# Producing runghc with a custom rule, as defined here, allows to use
+# it in rules which are using a Haskell toolchain for the target
+# platform.
+#
+
+def _runghc_impl(ctx):
+    hs_toolchain = ctx.toolchains["@rules_haskell//haskell:toolchain"]
+    (_, extension) = paths.split_extension(hs_toolchain.tools.runghc.path)
+
+    runghc_file = ctx.actions.declare_file(ctx.label.name + "2" + extension)
+    ctx.actions.symlink(
+        output = runghc_file,
+        target_file = hs_toolchain.tools.runghc,
+        is_executable = True,
+    )
+
+    return [DefaultInfo(
+        executable = runghc_file,
+        runfiles = hs_toolchain.cc_wrapper.runfiles.merge(
+            ctx.runfiles(files = [hs_toolchain.tools.runghc]),
```

```suggestion
            ctx.runfiles(files = [runghc_file, hs_toolchain.tools.runghc]),
```
See https://groups.google.com/g/bazel-discuss/c/oEyEIr7vaF4/m/0E5QaK46AQAJ

---

I'm a tad confused. Why does CI work without adding `runghc_file` to the runfiles?

---

Ah, it is probably the way in which the output of the rule is depended upon.

---

`runghc` is added to the action inputs explicitly, https://github.com/tweag/rules_haskell/pull/1510/files#diff-908241729f5583f805b295fedcf5a975ae9f8099ee83fae2a08496f9ba78b847R353.
This change is only relevant if you have another executable target that adds this runghc target to its `data` dependencies.